src/EcdsaProvider.ts

```diff
@@ -20,61 +8,47 @@ import type {
   KeyFormat,
   KeyUsage,
 } from './types'
+import { AskarCryptoKey, assertIsAskarCryptoKey } from './CryptoKey'
 
 export class EcdsaProvider extends core.EcdsaProvider {
-  public async onSign(algorithm: EcdsaParams, key: core.CryptoKey, data: ArrayBuffer): Promise<ArrayBuffer> {
-    if (algorithm.hash.name !== 'SHA-256') {
-      throw new Error(`Invalid hashing algorithm. Expected: 'SHA-256', received: ${algorithm.hash.name}`)
-    }
-
-    const signature = askarKeySign({ key, data })
-
-    return signature
+  public async onSign(algorithm: EcdsaParams, key: AskarCryptoKey, data: ArrayBuffer): Promise<ArrayBuffer> {
+    assertIsAskarCryptoKey(key)
+    return key.sign(algorithm, data)
   }
 
   public async onVerify(
     algorithm: EcdsaParams,
-    key: core.CryptoKey,
+    key: AskarCryptoKey,
     signature: ArrayBuffer,
     data: ArrayBuffer
   ): Promise<boolean> {
-    if (algorithm.hash.name !== 'SHA-256') {
-      throw new Error(`Invalid hashing algorithm. Expected: 'SHA-256', received: ${algorithm.hash.name}`)
-    }
-
-    const isValid = askarKeyVerify({ key, data, signature })
-
-    return isValid
+    assertIsAskarCryptoKey(key)
```

Comment:
This makes sense 👍 

I think if we make it a generic one, we can instead of making it an AskarCryptoKey, link it with a `Key` instance from Credo. Maybe we need to extend Key to also indicate usages (so we know if it's for verification or also signing).

Reply:
This class is not used anymore in the version with credo. That would be the `EcdsaCallbackProvider.ts` I left this in so that if there were issues, I would still hav the code.